src/EnergyPlus/EcoRoofManager.cc

```diff
@@ -931,11 +961,23 @@ namespace EcoRoofManager {
             // Smart schedule only irrigates when scheduled AND the soil is less than 40% saturated
             state.dataEcoRoofMgr->CurrentIrrigation = state.dataWaterData->Irrigation.ScheduledAmount; // units of m
             state.dataWaterData->Irrigation.ActualAmount = state.dataEcoRoofMgr->CurrentIrrigation;
+        } else if (state.dataWaterData->Irrigation.ModeID == DataWater::RainfallMode::Invalid) {
+            // no schedule, irrigation is just the rain amount
+            state.dataEcoRoofMgr->CurrentIrrigation = state.dataEcoRoofMgr->CurrentPrecipitation; // units of m
+            state.dataWaterData->Irrigation.ActualAmount = state.dataEcoRoofMgr->CurrentIrrigation;
         }
 
         Moisture += state.dataEcoRoofMgr->CurrentIrrigation / state.dataEcoRoofMgr->TopDepth; // irrigation in (m)/timestep put into top layer
         if (!state.dataGlobal->WarmupFlag) {
             state.dataEcoRoofMgr->CumIrrigation += state.dataEcoRoofMgr->CurrentIrrigation;
+            // aggregate to monthly for reporting
+            int EndYear = state.dataEnvrn->EndYear;
+            int CurrentYear = state.dataEnvrn->Year;
+            // only report for the last year
+            if (CurrentYear == EndYear) {
+                int month = state.dataEnvrn->Month;
+                state.dataEcoRoofMgr->MonthlyIrrigation[month - 1] += state.dataWaterData->Irrigation.ActualAmount * 1000.0;
```

Comment:
Use `if (state.dataEnvrn->EndYearFlag)` It's simpler, and I'm not sure `if (CurrentYear == EndYear)` is the correct test anyway. See how `EndYearFlag` is set [here](https://github.com/NREL/EnergyPlus/blob/fadc5973b85c70e8cc923efb69c144e808a26078/src/EnergyPlus/WeatherManager.cc#L1916-L1920).

Reply:
This is intended to make sure the table only reports data for the last year when the simulation spans multiple years (the behavior of other monthly tables). For a simulation period of 2010/1/1 to 2013/1/1, If I use EndYearFlag here, the table will aggregate data for 2011, 2012, and 2013, which is not what I wanted. I'll look into other methods. I'll also look into using the WriteMonthlyTables function instead for the reporting (which seems to automatically resets to last year).